cmd/monaco/download/download_command.go

```diff
@@ -253,3 +260,19 @@ func setupSharedFlags(cmd *cobra.Command, project, outputFolder *string, forceOv
 		log.Fatal("failed to setup CLI %v", err)
 	}
 }
+
+// printUploadToSameEnvironmentWarning function may display a warning message on the console,
+// notifying the user that downloaded objects cannot be uploaded to the same environment.
+// It verifies the version of the tenant and, depending on the result, it may or may not display the warning.
+func printUploadToSameEnvironmentWarning(environmentURL, token string) {
```

Comment:
for me dominant function is `check`, not `printWarning`.
That's why I would go with a name similar to `checkVersionCompatibility`

Reply:
disagree, the actual purpose is to warn the user by printing something on the screen. How it is done is described in the comment.